src/main/java/org/dataone/hashstore/filehashstore/FileHashStoreUtility.java

```diff
@@ -112,17 +116,75 @@ public static boolean isDirectoryEmpty(Path directory) throws IOException {
             // findFirst(). If the Optional contains a value (i.e., an element was found),
             // isPresent() returns true. If the Optional is empty (i.e., the stream is
             // empty), isPresent() returns false.
-            return !stream.findFirst().isPresent();
+            return stream.findFirst().isPresent();
         }
     }
 
     /**
-     * Checks whether a given string is empty and throws an exception if so
+     * Checks a directory for files and returns a list of paths
+     *
+     * @param directory Directory to check
+     * @return List<Path> of files
+     * @throws IOException If I/O occurs when accessing directory
+     */
+    public static List<Path> getFilesFromDir(Path directory) throws IOException {
+        List<Path> filePaths = new ArrayList<>();
+        if (Files.isDirectory(directory) && dirContainsFiles(directory)) {
+            try (Stream<Path> stream = Files.walk(directory)) {
+                stream.filter(Files::isRegularFile).forEach(filePaths::add);
+            }
+        }
+        return filePaths;
+    }
+
+    /**
+     * Rename the given path to the 'file name' + '_delete'
+     * 
+     * @param pathToRename The path to the file to be renamed with '_delete'
+     * @return Path to the file with '_delete' appended
+     * @throws IOException Issue with renaming the given file path
+     */
+    public static Path renamePathForDeletion(Path pathToRename) throws IOException {
+        ensureNotNull(pathToRename, "pathToRename", "renamePathForDeletion");
+        if (!Files.exists(pathToRename)) {
+            String errMsg = "FileHashStoreUtility.renamePathForDeletion - Given path to file: "
+                + pathToRename + " does not exist.";
+            throw new FileNotFoundException(errMsg);
+        }
+        Path parentPath = pathToRename.getParent();
+        Path fileName = pathToRename.getFileName();
+        String newFileName = fileName.toString() + "_delete";
+
+        Path deletePath = parentPath.resolve(newFileName);
+        Files.move(pathToRename, deletePath, StandardCopyOption.ATOMIC_MOVE);
+        return deletePath;
+    }
+
+    /**
+     * Delete all paths found in the given List<Path> object.
+     *
+     * @param deleteList Directory to check
+     * @throws IOException Unexpected I/O error when deleting files
+     */
+    public static void deleteListItems(Collection<Path> deleteList) throws IOException {
+        ensureNotNull(deleteList, "deleteList", "deleteListItems");
+        if (!deleteList.isEmpty()) {
+            for (Path deleteItem : deleteList) {
+                if (Files.exists(deleteItem)) {
```

Comment:
May we used the try-catch block there so we can delete as many as possible files?

Reply:
I have made this change, thank you!